PGM/src/main/java/tc/oc/pgm/commands/MapCommands.java

```diff
@@ -41,24 +51,101 @@
+        String gamemode = args.getFlag('g');
+        String author = args.getFlag('a');
+        if(args.getFlag('a') != null) {
+            syncExecutor.callback(
+                userFinder.findUser(sender, author, UserFinder.Scope.ALL, UserFinder.Default.NULL),
+                CommandFutureCallback.onSuccess(sender, result -> {
+                        BaseComponent displayMsg = args.hasFlag('g') ? new TranslatableComponent("command.map.mapList.displayByBoth",
+                            new PlayerComponent(identityProvider.currentIdentity(result.user)),
+                            new UserTextComponent(senderIdentity, gamemode.toLowerCase())) :
+                            new TranslatableComponent("command.map.mapList.displayByAuthor", new PlayerComponent(identityProvider.createIdentity(result.user)));
+                        displayMaps(sender,
+                            displayMsg,
+                            getFilteredMaps(parseGamemode(gamemode), result.user),
+                            args);
+                    }
+                ));
+        } else if(gamemode != null) {
+            displayMaps(sender,
+                new TranslatableComponent("command.map.mapList.displayByGamemode", new UserTextComponent(senderIdentity, gamemode.toLowerCase())),
+                getFilteredMaps(parseGamemode(gamemode), null),
+                args);
+        } else {
+            displayMaps(sender, new TranslatableComponent("command.map.mapList.title"), getFilteredMaps(null, null), args);
+        }
+    }
 
-        new PrettyPaginatedResult<PGMMap>(PGMTranslations.get().t("command.map.mapList.title", sender)) {
-            @Override public String format(PGMMap map, int index) {
-                return (index + 1) + ". " + map.getInfo().getShortDescription(sender);
+    private MapDoc.Gamemode parseGamemode(@Nullable String gamemode) throws CommandException {
+        if (gamemode == null) {
+            return null;
+        }
+        try {
+            return MapDoc.Gamemode.valueOf(gamemode);
+        } catch (IllegalArgumentException e) {
+            throw new CommandException("Invalid input " + gamemode);
+        }
+    }
+
+    /**
+     * Returns a list of filtered map with the given parameters.
+     * If both parameters are null, this method returns the full set of maps from the repository.
+     *
+     * @param gamemode Gamemode used to filter the map with - can be null.
+     * @param author   An author given from the result of a UserFinder - can be null.
+     */
+    private Set<PGMMap> getFilteredMaps(@Nullable MapDoc.Gamemode gamemode, @Nullable User author) {
+        final Set<PGMMap> maps = ImmutableSortedSet.copyOf(new PGMMap.DisplayOrder(), PGM.get().getMapLibrary().getMaps());
+        if(gamemode == null && author == null) {
+            return maps;
+        }
+        return maps.stream().filter(m -> gamemode == null || m.getDocument().gamemode().contains(gamemode))
```

Comment:
do not force 2 filters on the stream, you don't need them do one with an && `m -> (gm == null || m.doc.gm.contains(gm)) && (author == null || m.doc.aut.contains(author))`